swap_meet/vendor.py

```diff
@@ -1,2 +1,103 @@
+
+
 class Vendor:
-    pass
\ No newline at end of file
+    #
+    def __init__(self, inventory=None):
+        # if inventory is truthy, values will be assigned to inventory
+        if inventory:
+            self.inventory = inventory
+            # otherwise, []
+        else:
+            self.inventory = []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+# if theres a match, item gets removed
+    def remove(self, item):
+
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+
+        else:
+            return False
+
+# chhecking if id matches any in the invetory
+#
+    def get_by_id(self, id):
+        for each_item in self.inventory:
+            if each_item.id == id:
+                return each_item
+
+        return None
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        # if either of these are not found, returns False
+        if my_item not in self.inventory:
+            return False
+
+        if their_item not in other_vendor.inventory:
+            return False
+# if found
+# performs the swap
+        self_item_idx = self.inventory.index(my_item)
+        their_item_idx = other_vendor.inventory.index(their_item)
+# using a temp variable to hold the value then swap values
+        temp = self.inventory[self_item_idx]
+        self.inventory[self_item_idx] = their_item
+        other_vendor.inventory[their_item_idx] = temp
```

Comment:
Instead of using index to add/remove items, it would be better to reuse the `add` and `remove` methods you created earlier.